msal/application.py

```diff
@@ -1163,9 +1163,10 @@ def _sign_out(self, home_account):
                     and rt.get("family_id") == app_metadata["family_id"])
                 ]:
             self.token_cache.remove_rt(rt)
-        for at in self.token_cache.find(  # Remove ATs
+        for at in list(self.token_cache.search(  # Remove ATs from a snapshot
```

Comment:
I understand the reason of using `list()` is to avoid changing `self.token_cache` while it is being iterated, but it will be good to add a more detailed comment, otherwise the reader may not know why `list()` is used.

Reply:
Have we considered using a [`Lock`](https://docs.python.org/3/library/threading.html#lock-objects) to avoid such scenario?

Reply:
We discussed this offline. `Lock` is not a good choice here because:
1. Internally, TokenCache's methods may call each other and they need a `RLock` to coordinate with each other.
2. In our usage here, the default TokenCache is an in-memory `dict` which would automatically throw an error if we forgot to apply the `list(...)`.

```
  File "/home/demo/Live-usb-storage/repo/microsoft-authentication-library-for-python/msal/application.py", line 1209, in _sign_out
    for at in self.token_cache.search(  # Remove ATs from a static list,
  File "/home/demo/Live-usb-storage/repo/microsoft-authentication-library-for-python/msal/token_cache.py", line 147, in search
    for entry in self._cache.get(credential_type, {}).values():
RuntimeError: dictionary changed size during iteration
```